perf_guide.md

```diff
@@ -0,0 +1,149 @@
+# `std::simd`'s performance guide
```

Comment:
@GabrielMajeri ideally we would collect all the "common wisdom" about writing fast code with the library in this document before we forget about it.

Reply:
Sounds great, will see what I can contribute as soon as I find time. Should this contain SIMD-related stuff only, or should it also contains some related common sense things like "use a profiler, don't guess" or "avoid too many allocations"?

Reply:
> SIMD-related stuff only,

I'd like to focus this as much as possible on avoiding performance pitfalls when using `packed_simd`. I think that for all other stuff the documentation-wg wanted to start a book. If that ever happens this could be a chapter in that book about SIMD-related stuff.